reader.go

```diff
@@ -59,6 +59,16 @@ type Reader struct {
 	lag     int64
 	closed  bool
 
+	// Without a group subscription (when Reader.config.GroupID == ""),
+	// when errors occur, the Reader gets a synthetic readerMessage with
+	// a non-nil err set. With group subscriptions however, when an error
+	// occurs in Reader.run, there's no reader running (sic, cf. reader vs.
+	// Reader) and there's no way to let the high-level methods like
+	// FetchMessage know that an error indeed occurred. If an error in run
+	// occurs, it will be non-block-sent to this unbuffered channel, where
+	// the high-level methods can select{} on it and notify the caller.
+	runError chan error
```

Comment:
I think it would be a good idea to make this a buffered channel of capacity 1.  Otherwise, there would be a decent chance of dropping the error.  In the case where e.g. the connect string is bad, it's likely that the caller is blocking on the `Reader` and the synchronous handoff via `runError` will work.  If something goes awry after messages have already started flowing, though, it's less likely that the client will be blocking because it may be either be processing a message or the buffered `r.msgs` channel will have messages queued.

Reply:
My line of thinking was that this way, you'll only hit the error if it actually affects you, meaning you don't have anything to process, and you're not able to connect. E.g., if network was choppy but healed while you were processing a message, you will now (always) get the past error, though it's passé. When the channel isn't buffered, that won't happen.

But I agree that when you're in a known-wrong state (i.e., you exceeded your attempts inside Reader.run) it will take ReaderConfig.MaxAttempts (times the respective time-outs per attempt) to learn about that.

Perhaps a non-blocking read from buffered runError channel upon succeeding in Reader.run would be a good compromise between learning of the error promptly while not seeing past errors which are irrelevant?

I will address this plus the test tomorrow.

Reply:
It's an interesting case.  I wonder...if your reader can't maintain its membership with the consumer group, should you even be processing messages in the first place even if they're already buffered?  I would expect that the reader would have lost its assignments and that they'd have been given to another consumer already, so you may run the risk of duplicate processing because you wouldn't be able to commit offsets.